### PR TITLE
Add wallet TLS setup doc

### PR DESCRIPTION
**Pull request by hashmap**

### CODE REVIEW

File: doc/wallet/tls-setup.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## I don't have a TLS certificate
+You can get it for free from [Let's Encrypt](https://letsencrypt.org/). To simplify the process we need `certbot`.
+
+### Install certbot
+Go to [Certbot home page](https://certbot.eff.org/), choose I'm using `None of the above` and your OS (eg `Ubuntu 18.04` which will be used as an example). You will be redirected to a page with instructions like [steps for Ubuntu](https://certbot.eff.org/lets-encrypt/ubuntubionic-other). Follow instructions from `Install` section. As result you should have `certbot` installed.
+
+### Obtain certificate
+If you have experince with `certboot` feel free to use any type of the challenge, this guide covers the simplest case of HTTP challenge. For that you need to have a web server listening on port `80`, which requires running it as root in the simplest case. So there is no value to implement certbot authenticator in Grin even it runs web server. We will use a server provided by certbot. **Make sure you have port 80 open**
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion
If you have experince with `certboot` feel free to use any type of challenge. This guide covers the simplest case of HTTP challenge. For this you need to have a web server listening on port `80`, which requires running it as root in the simplest case. We will use the server provided by certbot. **Make sure you have port 80 open**
```

File: doc/wallet/tls-setup.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ 	 Your cert will expire on 2019-01-16. To obtain a new or tweaked
+ 	 version of this certificate in the future, simply run certbot
+ 	 again. To non-interactively renew *all* of your certificates, run
+ 	"certbot renew"
+```
+
+### Change permissions
+Now you have the certificate files but only root user can read it. We run grin as `ubuntu` user. There are different scenarios how to fix it, the simplest one is to create a group which will have access to `/etc/letsencrypt` directory and add our user to this group.
+
+* `$ sudo groupadd tls-cert`
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion

```
$ sudo groupadd tls-cert
$ sudo usermod -a -G tls-cert ubuntu
$ chgrp -R tls-cert /etc/letsencrypt
$ chmod -R g=rX /etc/letsencrypt
$ sudo chmod 2755 /etc/letsencrypt
```

```

File: doc/wallet/tls-setup.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+* `$ sudo usermod -a -G tls-cert ubuntu`
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion
```

File: doc/wallet/tls-setup.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+* `$ chgrp -R tls-cert /etc/letsencrypt`
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion
```

File: doc/wallet/tls-setup.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+* `$ chmod -R g=rX /etc/letsencrypt`
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion
```

File: doc/wallet/tls-setup.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+* `$ sudo chmod 2755 /etc/letsencrypt`
```
Comment (lehnberg) [inline doc/wallet/tls-setup.md]:
```suggestion
```

File: doc/wallet/tls-setup.md (hunk 2 above)
Comment (lehnberg) [reply doc/wallet/tls-setup.md]: Hmm, the suggestion is to put all the terminal command in one single code bracket, not sure if that worked properly.

Comment (hashmap) [reply doc/wallet/tls-setup.md]: @lehnberg just to be sure I made the suggested change manually, thanks! The first suggestion worked nicely.

**Pull request merged**